generator/internal/discovery_to_proto.h

```diff
@@ -94,6 +94,7 @@ Status GenerateProtosFromDiscoveryDoc(
     std::string const& protobuf_proto_path,
     std::string const& googleapis_proto_path, std::string const& output_path,
     std::string const& export_output_path,
+    bool enable_parallel_write_for_discovery_protos = false,
```

Comment:
Q.  Is it weird that the default value of the parameter is not the same as the default value for the flag?  Do we need a defaulted parameter at all?

Reply:
It is weird. They are different because the unit test calls this directly (which we don't want to be ||), but the main generator (which we would want to be || uses the CL arg)

Reply:
> the unit test calls this directly (which we don't want to be ||)

Then I would suggest that the unit test pass an explicit `/*enable_parallel_write_for_discovery_protos=*/false`.

Reply:
That's fair. I'll draw up the PR.

Reply:
https://github.com/googleapis/google-cloud-cpp/pull/13907